plenty_api/api.py

```diff
@@ -705,6 +705,66 @@ def plenty_api_get_variations(self,
             query=query,
             lang=lang)
 
+    def plenty_api_get_stock(self,
+                             refine: dict = None,
+                             lang: str = ''):
+        return self.plenty_api_generic_get(
+            domain='stockmanagement',
+            refine=refine,
+            lang=lang)
+
+    def plenty_api_get_storagelocations(self,
+                                        warehouse_id: int,
+                                        refine: dict = None,
+                                        additional: list = None,
+                                        lang: str = ''):
+        return self.plenty_api_generic_get(
+            domain='warehouses',
+            path=f'/{warehouse_id}/stock/storageLocations',
+            refine=refine,
+            additional=additional,
+            lang=lang)
+
+    def plenty_api_get_variation_stock_batches(self, variation_id):
+        # get all warehouses for this item
+        refine = {'variationId': variation_id}
+
+        # returns only locations with positive stock
+        stock = self.plenty_api_get_stock(refine=refine)
+        warehouses = [s['warehouseId'] for s in stock]
+
+        # get storage data from everywhere
+        # todo – think about romoving non-essenital keys
+        storage_data = [location
+                        for warehouse_id in warehouses
+                        for location in self.plenty_api_get_storagelocations(
+                        warehouse_id, refine=refine)]
+
+        # return ordered by best before date (oldest first)
+        return sorted(storage_data, key=lambda s: s['bestBeforeDate'])
+
+    def plenty_api_get_variation_warehouses(self, item_id, variation_id):
+        return self.plenty_api_generic_get(
+            domain='item',
+            path=f'/{item_id}/variations/{variation_id}/variation_warehouses')
+
+    def plenty_api_get_contacts(self,
+                                refine: dict = None,
+                                additional: list = None,
+                                lang: str = ''):
+        query = utils.sanity_check_parameter(domain='contact',
+                                             query={},
+                                             refine=refine,
+                                             additional=additional,
+                                             lang=lang)
+
+        contacts = self.__repeat_get_request_for_all_records(
+            domain='contact',
+            query=query)
+
+        return utils.transform_data_type(data=contacts,
+                                         data_format=self.data_format)
```

Comment:
I believe this can be handled as well with the `plenty_api_generic_get` wrapper. Was there something particular that stopped you from using it?

Reply:
That was an oversight, I fixed it now.